force-app/main/default/classes/JobApplicationHandlerTest.cls

```diff
@@ -112,4 +112,225 @@ public with sharing class JobApplicationHandlerTest {
+                Decimal fedIncomeTax = 3429.98;
+                Decimal medicare = 452.40;
+                Decimal socialSecurity = 1934.40;
+                Decimal takeHome = 46183.22;
+                System.assert(jobApp.Federal_Income_Tax_Withheld__c >= fedIncomeTax - acceptableRange && jobApp.Federal_Income_Tax_Withheld__c <= fedIncomeTax + acceptableRange, 'Federal Income Tax Incorrect');
+                System.assert(jobApp.Social_Security_Withheld__c >= socialSecurity - acceptableRange && jobApp.Social_Security_Withheld__c <= socialSecurity + acceptableRange, 'Social Security Incorrect');
+                System.assert(jobApp.Medicare_Withheld__c >= medicare - acceptableRange && jobApp.Medicare_Withheld__c <= medicare + acceptableRange, 'Medicare Incorrect');
+                System.assert(jobApp.Take_Home_Amount__c >= takeHome - acceptableRange && jobApp.Take_Home_Amount__c <= takeHome + acceptableRange, 'Take-Home Incorrect');
+            }
+            if(jobApp.Salary__c == 100 && jobApp.Salary_Type__c == 'Hourly' && jobApp.Tax_Filing_Status__c == 'Head of Household'){
+                Decimal fedIncomeTax = 34513.86;
+                Decimal medicare = 2581;
+                Decimal socialSecurity = 9932.40;
+                Decimal takeHome = 160972.74;
+                System.assert(jobApp.Federal_Income_Tax_Withheld__c >= fedIncomeTax - acceptableRange && jobApp.Federal_Income_Tax_Withheld__c <= fedIncomeTax + acceptableRange, 'Federal Income Tax Incorrect');
+                System.assert(jobApp.Social_Security_Withheld__c >= socialSecurity - acceptableRange && jobApp.Social_Security_Withheld__c <= socialSecurity + acceptableRange, 'Social Security Incorrect');
+                System.assert(jobApp.Medicare_Withheld__c >= medicare - acceptableRange && jobApp.Medicare_Withheld__c <= medicare + acceptableRange, 'Medicare Incorrect');
+                System.assert(jobApp.Take_Home_Amount__c >= takeHome - acceptableRange && jobApp.Take_Home_Amount__c <= takeHome + acceptableRange, 'Take-Home Incorrect');
+            }
+            if(jobApp.Salary__c == 150000 && jobApp.Salary_Type__c == 'Annual' && jobApp.Tax_Filing_Status__c == null){
+                System.assertEquals(0, jobApp.Take_Home_Amount__c, 'Take-Home Pay Should be 0');
+                System.assertEquals(0, jobApp.Federal_Income_Tax_Withheld__c, 'Federal Income Tax Should be 0');
+                System.assertEquals(0, jobApp.Social_Security_Withheld__c, 'Social Security Should be 0');
+                System.assertEquals(0, jobApp.Medicare_Withheld__c, 'Medicare Should be 0');
+                
+            }
+
+        }
+    }
+        
+
+
+    @IsTest
+    static void testdeductionUpdates(){
+        List<Job_Application__c> jobAppList = new List<Job_Application__c>();
+        Job_Application__c newJobApp1 = new Job_Application__c(Deduction_Type__c = 'Standard Deduction', Itemized_Deduction_Amount__c = 10000);
```

Comment:
@triden92 same thing we could add to the TestDataFactory if you want to.

Reply:
@rileymeye1 I'm fine doing this one as well. Just need to take a look at the level of effor.